lib/client.js

```diff
@@ -1722,27 +1722,23 @@ MatrixClient.prototype.loginWithSAML2 = function(relayState, callback) {
 };
 
 /**
- * @param {module:client.callback} callback Optional.
- * @return {module:client.Promise} Resolves: TODO
- * @return {module:http-api.MatrixError} Rejects: with an error response.
+ * @param {string} redirectUrl URL to which to be redirect to after
+ * HS authenticates with CAS and issues login token
+ * Will redirect to homeserver to complete CAS login
```

Comment:
Wha-? I sort of understand but the grammar is weird. Perhaps:

```
@param {string} redirectUrl The URL to redirect to after the HS authenticates with CAS.
```

Taking into account my other comment, you may want to also do:

```
@return {string} The HS URL to hit to begin the CAS login process.
```